src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/SecurityHelpers.cs

```diff
@@ -187,6 +189,40 @@ internal static class SecurityHelpers
                 StringComparer.Ordinal,
                 "ReadXml");
 
+        /// <summary>
+        /// Determines if an operation is inside autogenerated code that's probably for a GUI app.
+        /// </summary>
+        /// <param name="operationAnalysisContext">Context for the operation in question.</param>
+        /// <param name="wellKnownTypeProvider"><see cref="WellKnownTypeProvider"/> for the operation's compilation.</param>
+        /// <returns>True if so, false otherwise.</returns>
+        /// <remarks>
+        /// This is useful for analyzing categorizing cases into different rules to represent different risks.
+        /// </remarks>
+        public static bool IsOperationInsideAutogeneratedCodeForGuiApp(
+            OperationAnalysisContext operationAnalysisContext,
+            WellKnownTypeProvider wellKnownTypeProvider)
+        {
+            INamedTypeSymbol? xmlReaderTypeSymbol =
+                wellKnownTypeProvider.GetOrCreateTypeByMetadataName(
+                    WellKnownTypeNames.SystemXmlXmlReader); INamedTypeSymbol? designerCategoryAttributeTypeSymbol =
+                wellKnownTypeProvider.GetOrCreateTypeByMetadataName(
+                    WellKnownTypeNames.SystemComponentModelDesignerCategoryAttribute);
+            INamedTypeSymbol? debuggerNonUserCodeAttributeTypeSymbol =
+                wellKnownTypeProvider.GetOrCreateTypeByMetadataName(
+                    WellKnownTypeNames.SystemDiagnosticsDebuggerNonUserCode);
+
+            return
+                operationAnalysisContext.ContainingSymbol?.MetadataName == "ReadXmlSerializable"
+                && operationAnalysisContext.ContainingSymbol?.DeclaredAccessibility == Accessibility.Protected
+                && operationAnalysisContext.ContainingSymbol?.IsOverride == true
```

Comment:
If the first condition succeed then `ContainingSymbol` cannot be `null`.

```suggestion
                && operationAnalysisContext.ContainingSymbol.DeclaredAccessibility == Accessibility.Protected
                && operationAnalysisContext.ContainingSymbol.IsOverride
```